### PR TITLE
github perm syncing: remove useless additional request when listing collaborators

### PR DESCRIPTION
**Pull request by mrnugget**
I noticed that when looking at "Outbound requests" in the site-admin panel that we send 2 requests to list collaborators, even if that repository is a private repository of mine with a single collaborator.

Since we list 100 collaborators per page - why would we need to send two requests to list 1 collaborator?

So I dug in and found that we always do a second request if we have >0 collaborators returned, which doesn't make a lot of sense.

This fix is just something I hacked together, there might (should?) be a better way to determine whether there is a next page.

## TODO

- [ ] Fix tests
- [ ] Find better solution

## Test plan

- Manual testing

### CODE REVIEW

File: internal/extsvc/github/v3.go [hunk 1]
```diff
@@ -440,7 +440,7 @@ func (c *V3Client) ListRepositoryCollaborators(ctx context.Context, owner, repo
 	if err != nil {
 		return nil, false, err
 	}
-	return users, len(users) > 0, nil
+	return users, len(users) == 100, nil
```
Comment (eseliger) [inline internal/extsvc/github/v3.go:443]: Can we make this 100 and the 100 in the query param a constant so that we won’t break this logic?

**Review by eseliger — APPROVED** [review]

**Comment by miveronese** [comment]
Awesome finding! 🎉

File: internal/extsvc/github/v3.go (hunk 1 above)
Comment (mrnugget) [reply internal/extsvc/github/v3.go:443]: Yeah, wanted to do that. But we @kopancek and @pjlast found a better to determine whether we have a next page or not: if there _is_ a next page then GitHub returns a `Link` header. So we can check for that.

This here is basically a fix and then we'll do a proper fix for this request and the 3 others that have wrong checks.

**Comment by pjlast** [comment]
https://github.com/sourcegraph/sourcegraph/pull/44955

**Comment by mrnugget** [comment]
closing in favor of https://github.com/sourcegraph/sourcegraph/pull/44955

**Pull request closed without merging**